selfservice/flow/login/handler_test.go

```diff
@@ -1,67 +1,91 @@
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 	"github.com/tidwall/gjson"
 
+	"github.com/ory/kratos/selfservice/form"
 	"github.com/ory/viper"
 
 	"github.com/ory/kratos/driver/configuration"
 	"github.com/ory/kratos/internal"
 	"github.com/ory/kratos/selfservice/errorx"
 	"github.com/ory/kratos/selfservice/flow/login"
-	"github.com/ory/kratos/selfservice/strategy/oidc"
-	"github.com/ory/kratos/selfservice/strategy/password"
 	"github.com/ory/kratos/session"
 	"github.com/ory/kratos/x"
 )
 
+type (
+	withTokenGenerator interface {
+		WithTokenGenerator(g form.CSRFGenerator)
+	}
+)
+
 func init() {
 	internal.RegisterFakes()
 }
 
-func TestEnsureSessionRedirect(t *testing.T) {
+func TestHandlerSettingReauth(t *testing.T) {
 	_, reg := internal.NewRegistryDefault(t)
+	for _, strategy := range reg.LoginStrategies() {
+		// We need to know the csrf token
+		strategy.(withTokenGenerator).WithTokenGenerator(x.FakeCSRFTokenGenerator)
+	}
 
 	router := x.NewRouterPublic()
 	admin := x.NewRouterAdmin()
 	reg.LoginHandler().RegisterPublicRoutes(router)
 	reg.LoginHandler().RegisterAdminRoutes(admin)
+	reg.LoginHandler().WithTokenGenerator(x.FakeCSRFTokenGenerator)
 	reg.LoginStrategies().RegisterPublicRoutes(router)
 	ts := httptest.NewServer(router)
 	defer ts.Close()
 
-	redirTS := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		_, _ = w.Write([]byte("already authenticated"))
-	}))
-	defer redirTS.Close()
+	loginTS := httptest.NewServer(login.TestRequestHandler(t, reg))
 
-	viper.Set(configuration.ViperKeyURLsDefaultReturnTo, redirTS.URL)
 	viper.Set(configuration.ViperKeyURLsSelfPublic, ts.URL)
+	viper.Set(configuration.ViperKeyURLsLogin, loginTS.URL)
 	viper.Set(configuration.ViperKeyDefaultIdentityTraitsSchemaURL, "file://./stub/login.schema.json")
 
-	for k, tc := range [][]string{
-		{"GET", login.BrowserLoginPath},
+	t.Run("does not set reauth flag on unauthenticated request", func(t *testing.T) {
+		c := ts.Client()
+		res, err := c.Get(ts.URL + login.BrowserLoginPath)
+		require.NoError(t, err)
+		defer res.Body.Close()
+		body, err := ioutil.ReadAll(res.Body)
 
-		{"POST", password.LoginPath},
+		assert.Equal(t, false, gjson.GetBytes(body, "is_reauthentication").Bool(), "%s", body)
+	})
 
-		// it is ok that these contain the parameters as raw strings as we are only interested in checking if the middleware is working
-		{"POST", oidc.AuthPath},
-		{"GET", oidc.AuthPath},
-		{"GET", oidc.CallbackPath},
-	} {
-		t.Run(fmt.Sprintf("case=%d/method=%s/path=%s", k, tc[0], tc[1]), func(t *testing.T) {
-			body, _ := session.MockMakeAuthenticatedRequest(t, reg, router.Router, x.NewTestHTTPRequest(t, tc[0], ts.URL+tc[1], nil))
-			assert.EqualValues(t, "already authenticated", string(body))
-		})
-	}
+	t.Run("does set reauth flag on authenticated request", func(t *testing.T) {
+		rid := x.NewUUID()
+		req := x.NewTestHTTPRequest(t, "GET", ts.URL+login.BrowserLoginPath, nil)
```

Comment:
This is missing `?prompt=true`

Reply:
it gets set here https://github.com/ory/kratos/pull/248/files#diff-318ba7d56b3e558e24dd7d812eb7bfccR80

Reply:
gotcha 👍